src/components/authentication/SignupForm.jsx

```diff
@@ -1,118 +1,40 @@
-    } else {
-      window.Notify.error('Passwords do not match');
-    }
-  }
-
-  signup() {
-    const { signUp } = this.props;
-    const user = { ...this.state };
-    signUp(user);
-  }
-
-  render() {
-    const { username, password, email, confirmPassword } = this.state;
-    return (
-      <form onSubmit={this.onSubmit}>
-        <div className="form-group">
-          <label htmlFor="username" className="has-input">
-            Username
-            <input
-              id="username"
-              type="text"
-              name="username"
-              value={username}
-              onChange={this.onChange}
-              className="form-control"
-              required
-            />
-          </label>
-        </div>
-
-        <div className="form-group">
-          <label htmlFor="email" className="has-input">
-            Email
-            <input
-              id="email"
-              type="email"
-              name="email"
-              value={email}
-              onChange={this.onChange}
-              className="form-control"
-              required
-            />
-          </label>
-        </div>
-
-        <div className="form-group">
-          <label htmlFor="password" className="has-input">
-            Password
-            <input
-              id="password"
-              type="password"
-              name="password"
-              value={password}
-              onChange={this.onChange}
-              className="form-control"
-              pattern="(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{8,}"
-              title="Must contain at least one number and 
-              one uppercase and lowercase letter, and at 
-              least 8 or more characters"
-              required
-            />
-          </label>
-        </div>
-
-        <div className="form-group">
-          <label htmlFor="confirmPassword" className="has-input">
-            confirm password
-            <input
-              id="confirmPassword"
-              type="password"
-              name="confirmPassword"
-              value={confirmPassword}
-              onChange={this.onChange}
-              className="form-control"
-            />
-          </label>
-        </div>
-
-        <div className="form-group">
-          <button type="submit" className="btn btn-primary btn-block">
-            Sign Up
-          </button>
-        </div>
-      </form>
-    );
-  }
-}
+import { Form, Input } from 'forms';
+
+const SignupForm = ({ onSuccess }) => (
+  <Form
+    name="login-form"
+    action="/users/"
+    button={() => (
+      <div className="form-group">
+        <button type="submit" className="btn btn-primary btn-block">
+          Sign Up
+        </button>
+      </div>
+    )}
+    beforeSubmit={(data, form) => {
+      if (data.password && data.password_confirmation === data.password) return { user: data };
+      form.updateErrors({ password_confirmation: ['Passwords do not match'] });
+      // don't submit
```

Comment:
remove commented line

Reply:
That comment  was aimed  at emphasizing to the future developer that the `return false ` prevents the form from being submitted .

Let me know if we can't have it there so that I delete it.